### ISSUE TITLE
Support .NET 5

### ISSUE THREAD

**Issue by HydrateEfficiently** [jwaliszko/ExpressiveAnnotations#90]
Hey Jaroslaw,

.NET 5 is being release soon and it will require quite a major migration to get this library working there. I would suggest creating a new Git repo to support 4 and 5.

It will also require some changes to the deployment. Microsoft is getting up with the play and delegating script deployment to bower/npm, so changes will be required here as well.

As far as actual code changes go, I don't believe there is much. Only the model state provider?

If you're willing to go forward with this and need a hand, I I'll try and free up some time.

Cheers,
Michael

**Comment by jwaliszko** [comment]
Hi. Thank you for your time and input. I'll take a closer look at this issue within the next few days, because currently I have no my real computer with me, only smartphone.
/jarek

**Comment by jwaliszko** [comment]
Excuse the delay. I'm currently working on C++ desktop apps and have no experience with ASP.NET 5 yet. I see that there are indeed heavy changes by reading this: http://docs.asp.net/en/latest/conceptual-overview/aspnet.html and that: http://blog.nuget.org/. I need some time to get through it. I'd like to stay with the same repository. Probably the packages will be different since there will be no JavaScript anymore inside ExpressiveAnnotations.nupkg. I'd like to find some recommended ways for upgrading cross-lang libs (C#/JS) from old to new ASP.NET stack, along with corresponding nuget packages.

**Comment by hatfies3** [comment]
Hi, we are just beginning to look at building out some ASP.Net 5/MVC6 applications.   Do you have any plans to build out a compatible library?

**Comment by jwaliszko** [comment]
@hatfies3: I have plans to do it when ASP.NET Core 1.0 (ASP.NET 5) become slightly more mature. There seems to be quite a lot of brainstorming at current stage of how things should be designed, and probably by the end of this year ([Q4 2016 / Q1 2017](https://blogs.msdn.microsoft.com/dotnet/2016/07/15/net-core-roadmap/)) I'll try to provide .NET Standard-based PCL (new substitute for old-style profile-based PCLs) for EA (probably with tiny, yet respective core version of web sample). If you need this earlier, fork the repo and do some experiments.

**Comment by hatfies3** [comment]
Thanks for coming back to me and confirming.   Our MVC6 plans are currently
shelved for the minute anyway but will pick this up later. I shall take a
look at whats involved if i get a chance.

**Comment by madrianr** [comment]
Hi, are there some news about ASP.NET NET Core 1.0 or 1.1 Support?
I use EA in many Projects and now we are startet a new one with ASP.NET Core 10.0/1.1 with NET Framework 4.6.2 and need a ASP.NET Core compatble version of EA...
robert

**Comment by jwaliszko** [comment]
Hi, unfortunately there are none. I'm out of time to take a look at this. I cannot say right now when it could be possibly done.

**Comment by madrianr** [comment]
Hi, that are bad News to me :-(
so we have the need to switch to another solution for ASP.NET Core

**Comment by jwaliszko** [comment]
@madrianr: if you have some spare time you can also fork the repository and try to port the library according to your needs untill the official version is provided.

**Comment by madrianr** [comment]
Hi, my problem is also that I have no time for doing that ;-)
what do you think, how much work will it be to migrate the library to work with ASP.NET Core?

**Comment by madrianr** [comment]
I used NET Portability Analyser to see about incompatibilities to ASP.NET Core (see attachement)
looks like a lot of work to do...
[ApiPortAnalysis.zip](https://github.com/jwaliszko/ExpressiveAnnotations/files/612861/ApiPortAnalysis.zip)

**Comment by jwaliszko** [comment]
@madrianr: you're right, it is probably a bit demanding, that's why I'd accept any help from the community side, e.g. some tips related to best practices, any respective sources, pull requests, etc. I have no experience with portable libraries development, that's why I cannot provide reliable estimation - especially if relevant sample project and unit tests are to be provided as well (which is a must for the release).

**Comment by madrianr** [comment]
Some News about ASP.NET Core Support?

**Comment by jwaliszko** [comment]
@madrianr: nope, no time for that. Unfortunately I cannot provide any concrete declaration of when it can be done.

**Comment by dougajmcdonald** [comment]
We have a project which uses EA and which we need to migrate to core, if I can get any time I will fork and have a play.
The cross language bit concerns me but the attributes should be simple enough.
